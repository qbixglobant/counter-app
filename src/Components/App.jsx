import PropTypes from "prop-types";

const App = ({ name, subtitle }) => {
  const h1 = "text-center mt-3 green display-1";
  const h2 = "text-center blue display-2";
  return (
    <>
      <h1 className={ h1 }>{ name }</h1>
      <h2 className={ h2 }>{ subtitle }</h2>
    </>
  );
};

App.propTypes = {
  name: PropTypes.string.isRequired
};

App.defaultProps = {
  name: "",
  subtitle: "Subtitle"
};

export default App;