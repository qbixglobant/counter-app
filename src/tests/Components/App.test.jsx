import { shallow } from "enzyme";
import App from "../../Components/App";

describe('Tests on <App /> Component', () => {
    // Default Testing
    // test('should show name on component', () => {
    //     const name = "John Doe";
    //     render( <App name={ name } /> );
    //     expect( screen.getByText(name) ).toBeInTheDocument();
    // });

    test('should snap with the snapshot', () => {
        const name = "John Doe";
        const wrapper = shallow( <App name={ name } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show given params on component', () => {
        const name = "John Doe";
        const subtitle = "Lorem Ipsum"
        const wrapper = shallow(
            <App name={ name } subtitle={ subtitle } />
        );
        const h1 = wrapper.find("h1").text();
        const h2 = wrapper.find("h2").text();

        expect( h1 ).toBe( name );
        expect( h2 ).toBe( subtitle );
    });

    test('should *', () => {
        const wrapper = shallow( <App /> );

        const h1 = wrapper.find("h1").text();

        expect( h1 ).toBe( "" );
    });

    test('should show default subtitle if subtitle param is not present', () => {
        const name = "John Doe";
        const wrapper = shallow(
            <App name={ name } />
        );

        const h2 = wrapper.find("h2").text();

        expect( h2 ).toBe( "Subtitle" );
    });
});
