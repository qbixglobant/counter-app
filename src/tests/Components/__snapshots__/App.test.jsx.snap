// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <App /> Component should snap with the snapshot 1`] = `
<Fragment>
  <h1
    className="text-center mt-3 green display-1"
  >
    John Doe
  </h1>
  <h2
    className="text-center blue display-2"
  >
    Subtitle
  </h2>
</Fragment>
`;
